src/cargo/util/toml/mod.rs

```diff
@@ -218,10 +218,32 @@ fn warn_on_unused(unused: &BTreeSet<String>, warnings: &mut Vec<String>) {
     }
 }
 
+pub fn prepare_for_publish(me: &Package, ws: &Workspace<'_>) -> CargoResult<Package> {
```

Comment:
While not necessary, I feel like `prepare_for_publish` and friends could be moved to a dedicated module.

Reply:
I've been going back and forth on it, so I've just left it for now.

The main benefit for keeping it here is to keep awareness of it when adding new features to the manifest in the hope that we remember that they might impact this so we update it.